common/workunit/workunit.cpp

```diff
@@ -9016,6 +9020,28 @@ void CLocalWorkUnit::setStatistic(StatisticCreatorType creatorType, const char *
     }
     if (kind == StCostCompile)
         p->setPropInt64("@costCompile", value);
+
+    // Special case - update aggregates for dfu FileAccessCost.  This is needed because although
+    // fileservices can update dfu cost in the workunit, it does not have a mechanism to update
+    // the aggregates for this stat (i.e. fileservices doesn't have access to a StatsAggregator.
+    // And say the StatsAggregator were created here, it would need to load all the global stats
+    // from dali,  mark the stat as 'dirty', and then have the aggregator refresh the aggregates.
+    // This would be slow and unnessarily cumbersome.) Note also that if the stat is written
+    // with the expectation that the hThor/Thor stats aggregator would generate the aggregate for
+    // this stat, it would not be the case as the aggregates are only created for stats marked
+    // 'dirty'.
```

Comment:
The comment update wasn't quite what I had in mind. 
I've opened a new JIRA to potentially reviist this: https://hpccsystems.atlassian.net/browse/HPCC-31703

> And say the StatsAggregator were created here, it would need to load all the global stats from dali

Each filespray service call is called from a context that already has a CStatsAggregator, and has already loaded the existing stats, and has the ability to mark target scopes dirty.
However, fileservices doesn't have access to the in flight 'statsAggregator' (perhaps could have via ICodeContext).

And, CStatsAggregator would need an alternate 'recordStats' mechanism that took an arbitrary scope, and possible a CRuntimeStatisitcsCollections for the specific stats being updated.

The advantage would be that it would share the same rollup aggregator mechanism as used elsewhere, and it would allow the stat. updates vs the aggregation and publication to be done at different times (e.g. is spraying lots of small files, the updates to Dali constantly could become expensive.